### PR TITLE
Activemq

### PR DESCRIPTION
**Pull request by armstrmi**
**Description:**

* added in Metric receiver for activemq

**Existing Issue(s):**

**Testing:**

Added integration tests for activemq metrics

**Documentation:**

* Documentation for activemq metrics added

### CODE REVIEW

File: jmx-metrics/docs/target-systems/activemq.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# ActiveMQ Metrics
+
+The JMX Metric Gatherer provides built in Tomcat metric gathering capabilities.
```
Comment (Mrod1598) [inline jmx-metrics/docs/target-systems/activemq.md:3]: Update this to say ActiveMQ
```suggestion
The JMX Metric Gatherer provides built in ActiveMQ metric gathering capabilities.
```

**Review by Mrod1598 — CHANGES_REQUESTED** [review]

**Comment by trask** [comment]
hi @armstrmi! can you fetch `upstream and merge `upstream/main` into your PR? that should fix the "Component Owners" build failure until we can fix the underlying github action (https://github.com/dyladan/component-owners/pull/10)

File: jmx-metrics/docs/target-systems/activemq.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+
+* Name: `activemq.message.dequeued`
+* Description: The total number of messages delivered to consumers.
+* Unit: `messages`
+* Labels: `destination`
+* Instrument Type: LongCounterCallback
+
+
+* Name: `activemq.wait_time.avg`
```
Comment (Mrod1598) [inline jmx-metrics/docs/target-systems/activemq.md:75]: looks like this name is wrong compared to the script `activemq.message.wait_time.avg`

File: jmx-metrics/docs/target-systems/activemq.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# ActiveMQ Metrics
```
Comment (rmfitzpatrick) [inline jmx-metrics/docs/target-systems/activemq.md:1]: Mind adding a link to this in the main table https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/README.md#target-systems?

File: jmx-metrics/docs/target-systems/activemq.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+The JMX Metric Gatherer provides built in ActiveMQ metric gathering capabilities.
+These metrics are sourced from: https://activemq.apache.org/jmx
+
+### Metrics
+
+* Name: `activemq.consumer.count`
+* Description: The number of consumers currently reading from the broker.
+* Unit: `{consumers}`
+* Labels: `destination`
+* Instrument Type: LongUpDownCounterCallback
```
Comment (rmfitzpatrick) [inline jmx-metrics/docs/target-systems/activemq.md:12]: We need to standardize these names throughout our docs, but should these be something like `ObservableLongUpDownCounter`? I'm not sure where the spec has landed.

File: jmx-metrics/docs/target-systems/activemq.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# ActiveMQ Metrics
+
+The JMX Metric Gatherer provides built in ActiveMQ metric gathering capabilities.
+These metrics are sourced from: https://activemq.apache.org/jmx
+
+### Metrics
+
+* Name: `activemq.consumer.count`
+* Description: The number of consumers currently reading from the broker.
+* Unit: `{consumers}`
```
Comment (rmfitzpatrick) [inline jmx-metrics/docs/target-systems/activemq.md:10]: nit, is `consumers` ok or are the braces meaningful?

File: jmx-metrics/src/main/resources/target-systems/activemq.groovy [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    "org.apache.activemq:type=Broker,brokerName=*,destinationType=Topic,destinationName=*"
+  ]
+)
+
+
+otel.instrument(activemqMetrics,
+  "activemq.producer.count",
+  "The number of producers currently attached to the broker.",
+  "{producers}",
+  ["destination" : { mbean -> mbean.name().getKeyProperty("destinationName") }],
```
Comment (rmfitzpatrick) [inline jmx-metrics/src/main/resources/target-systems/activemq.groovy:30]: Can we also set the `brokerName` as a label/attribute? I think it's applicable for all metrics here.

File: jmx-metrics/src/test/java/io/opentelemetry/contrib/jmxmetrics/JmxConfigTest.java [hunk 7]
```diff
@@ -16,7 +16,8 @@
   @Test
   void staticValues() {
     assertThat(JmxConfig.AVAILABLE_TARGET_SYSTEMS)
-        .containsOnly("cassandra", "jvm", "kafka", "kafka-consumer", "kafka-producer", "tomcat");
+        .containsOnly(
+            "cassandra", "jvm", "kafka", "kafka-consumer", "kafka-producer", "tomcat", "activemq");
```
Comment (rmfitzpatrick) [inline jmx-metrics/src/test/java/io/opentelemetry/contrib/jmxmetrics/JmxConfigTest.java:20]: nit, alphabetical order is desirable here and below

**Review by Mrod1598 — CHANGES_REQUESTED** [review]

File: jmx-metrics/docs/target-systems/activemq.md (hunk 5 above)
Comment (armstrmi) [reply jmx-metrics/docs/target-systems/activemq.md:10]: not necessarily, that seems easy enough

File: jmx-metrics/src/main/resources/target-systems/activemq.groovy (hunk 6 above)
Comment (armstrmi) [reply jmx-metrics/src/main/resources/target-systems/activemq.groovy:30]: sure thing

File: jmx-metrics/src/test/java/io/opentelemetry/contrib/jmxmetrics/JmxConfigTest.java (hunk 7 above)
Comment (armstrmi) [reply jmx-metrics/src/test/java/io/opentelemetry/contrib/jmxmetrics/JmxConfigTest.java:20]: gotcha, will implement

File: jmx-metrics/docs/target-systems/activemq.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
+# ActiveMQ Metrics
+
+The [JMX Metric Gatherer]( https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/README.md#target-systems) provides built in ActiveMQ metric gathering capabilities.
```
Comment (armstrmi) [inline jmx-metrics/docs/target-systems/activemq.md:3]: https://github.com/open-telemetry/opentelemetry-java-contrib/pull/188/files#r784332550 @rmfitzpatrick I put that link in here is that alright?

File: jmx-metrics/docs/target-systems/activemq.md (hunk 4 above)
Comment (Mrod1598) [reply jmx-metrics/docs/target-systems/activemq.md]: I'm curious about this one, can't seem to find any reference to how we should be naming it, personally, to me, it seems to make sense to match what's actually being used in the script. Is there a reason we're adding "observable"? just wasn't sure about this one. Thanks for any clarification!

File: jmx-metrics/docs/target-systems/activemq.md (hunk 8 above)
Comment (rmfitzpatrick) [reply jmx-metrics/docs/target-systems/activemq.md:3]: No need to make a link here, the readme change is what I was requesting.

**Comment by armstrmi** [comment]
> 

@rmfitzpatrick  could you please take a look at this when you get the chance

File: jmx-metrics/docs/target-systems/activemq.md (hunk 4 above)
Comment (rmfitzpatrick) [reply jmx-metrics/docs/target-systems/activemq.md]: https://github.com/open-telemetry/opentelemetry-specification/blob/5d374f8045dcb24523f5f17a4b8ba94542fd0b70/specification/metrics/api.md#asynchronous-counter and the java api uses "Observable" for its measurement interfaces.

**Review by rmfitzpatrick — APPROVED** [review]

**Comment by rmfitzpatrick** [comment]
> > 
> 
> @rmfitzpatrick could you please take a look at this when you get the chance

Thanks for the addition!

**Pull request merged**